src/Database/QueryCompiler.php

```diff
@@ -344,35 +337,75 @@ protected function _buildSetPart(array $parts, Query $query, ValueBinder $binder
     }
 
     /**
-     * Builds the SQL string for all the UNION clauses in this query, when dealing
+     * Builds the SQL string for all the `operation` clauses in this query, when dealing
      * with query objects it will also transform them using their configured SQL
      * dialect.
      *
-     * @param array $parts list of queries to be operated with UNION
-     * @param \Cake\Database\Query $query The query that is being compiled
-     * @param \Cake\Database\ValueBinder $binder Value binder used to generate parameter placeholder
+     * @param string $operation
+     * @param array $parts
+     * @param \Cake\Database\Query $query
+     * @param \Cake\Database\ValueBinder $binder
      * @return string
      */
-    protected function _buildUnionPart(array $parts, Query $query, ValueBinder $binder): string
-    {
-        $parts = array_map(function ($p) use ($binder) {
+    protected function _buildSetOperationPart(
+        string $operation,
+        array $parts,
+        Query $query,
+        ValueBinder $binder
+    ): string {
+        $setOperationsOrderBy = $query
+            ->getConnection()
+            ->getDriver($query->getConnectionRole())
+            ->supports(DriverFeatureEnum::SET_OPERATIONS_ORDER_BY);
+
+        $parts = array_map(function ($p) use ($binder, $setOperationsOrderBy) {
             /** @var \Cake\Database\Expression\IdentifierExpression $expr */
             $expr = $p['query'];
             $p['query'] = $expr->sql($binder);
-            $p['query'] = str_starts_with($p['query'], '(') ? trim($p['query'], '()') : $p['query'];
+            $p['query'] = $p['query'][0] === '(' ? trim($p['query'], '()') : $p['query'];
```

Comment:
Please don't use `[]` for strings, it kills readability  the original `str_starts_with()` was good.

Reply:
yeah! it looks I had an outdated version of the buildUniontPart method..fixed!